localstripe/localstripe-v3.js

```diff
@@ -212,101 +309,14 @@ Stripe = (apiKey) => {
-          return {error: body.error};
-        } else if (body.status === 'succeeded') {
-          return {error: null, setupIntent: body};
-        } else if (body.status === 'requires_action') {
-          const url =
-            (await openModal('3D Secure\nDo you want to confirm or cancel?',
-                             'Complete authentication', 'Fail authentication'))
-            ? `${LOCALSTRIPE_SOURCE}/v1/setup_intents/${seti}/confirm`
-            : `${LOCALSTRIPE_SOURCE}/v1/setup_intents/${seti}/cancel`;
-          response = await fetch(url, {
-            method: 'POST',
-            body: JSON.stringify({
-              key: apiKey,
-              use_stripe_sdk: true,
-              client_secret: clientSecret,
-            }),
-          });
-          body = await response.json().catch(() => ({}));
-          if (response.status !== 200 || body.error) {
-            return {error: body.error};
-          } else if (body.status === 'succeeded') {
-            return {error: null, setupIntent: body};
-          } else {  // 3D Secure authentication cancelled by user:
-            return {error: {message:
-              'The latest attempt to set up the payment method has failed ' +
-              'because authentication failed.'}};
-          }
-        } else {
-          return {error: {message: `setup_intent has status ${body.status}`}};
-        }
-      } catch (err) {
-        if (typeof err === 'object' && err.error) {
-          return err;
-        } else {
-          return {error: err};
-        }
-      }
+    handleCardSetup: async (clientSecret, element, data) => {  // deprecated
+      return confirmCardSetup(clientSecret, {payment_method: {card: element}});
     },
-    handleCardPayment: async (clientSecret, element, data) => {
-      console.log('localstripe: Stripe().handleCardPayment()');
-      try {
-        const success = await openModal(
-          '3D Secure\nDo you want to confirm or cancel?',
-          'Complete authentication', 'Fail authentication');
-        const pi = clientSecret.match(/^(pi_\w+)_secret_/)[1];
-        const url = `${LOCALSTRIPE_SOURCE}/v1/payment_intents/${pi}` +
-                    `/_authenticate?success=${success}`;
-        const response = await fetch(url, {
-          method: 'POST',
-          body: JSON.stringify({
-            key: apiKey,
-            client_secret: clientSecret,
-          }),
-        });
-        const body = await response.json().catch(() => ({}));
-        if (response.status !== 200 || body.error) {
-          return {error: body.error};
-        } else {
-          return {paymentIntent: body};
-        }
-      } catch (err) {
-        if (typeof err === 'object' && err.error) {
-          return err;
-        } else {
-          return {error: err};
-        }
-      }
+    confirmCardSetup: confirmCardSetup,
+    handleCardPayment: async (clientSecret, element, data) => {  // deprecated
+      return confirmCardPayment(clientSecret);
```

Comment:
Isn't it needed to pass some `data` here?

Reply:
`data` is part of the function prototype, but it's not use.
It was already like that before, and it's the case for `handleCardSetup` where data is ignored:
```js
    handleCardSetup: async (clientSecret, element, data) => {  // deprecated
      return confirmCardSetup(clientSecret, {payment_method: {card: element}});
    },
```
Do you think it's a bad idea?

Reply:
OK! Let's ignore it.